scrapy/crawler.py

```diff
@@ -69,6 +70,19 @@ def __init__(self, spidercls, settings=None):
 
         lf_cls = load_object(self.settings['LOG_FORMATTER'])
         self.logformatter = lf_cls.from_crawler(self)
+
+        if init_reactor:
+            # this needs to be done after the spider settings are merged,
+            # but before something imports twisted.internet.reactor
+            if self.settings.get("TWISTED_REACTOR"):
+                install_reactor(self.settings["TWISTED_REACTOR"], self.settings["ASYNCIO_EVENT_LOOP"])
+            else:
+                from twisted.internet import default
+                default.install()
+            log_reactor_info()
+        if self.settings.get("TWISTED_REACTOR"):
+            verify_installed_reactor(self.settings["TWISTED_REACTOR"])
```

Comment:
works fine, tested this in my project. Minor thing, you could probably simplify this to something like this to avoid repeatedly calling self.settings.get('TWISTED_REACTOR')

```python
reactor_class = self.settings.get("TWISTED_REACTOR")
if init_reactor:
    if reactor_class:
        install_reactor(reactor_class, self.settings["ASYNCIO_EVENT_LOOP"])
... etc

```

I also wonder if default value for init_reactor should be False? Are there any legitimate cases when you don't want to init reactor if you have it in settings?

Reply:
Thanks, I'll implement this simplification. As for the flag, it was the default behavior before this change, and the use case to not init the reactor is CrawlerRunner.